resources/session02/homework/http_server.py

```diff
@@ -33,9 +39,24 @@ def parse_request(request):
     return uri
 
 
+def handle_directory(uri):
+    return ("\n".join(listdir(uri)).encode("utf8"), "text/plain".encode("utf8"))
+
+def handle_file(uri):
+    with open (uri, "rb") as file:
+        contents = file.read()
+        mime_type, _ =  mimetypes.guess_type(uri)
+        return contents, mime_type.encode('utf8')
+
+
 def resolve_uri(uri):
-    """This method should return appropriate content and a mime type"""
-    return b"still broken", b"text/plain"
+    cwd = getcwd() 
+    absolute_path =  os.path.join(cwd, "webroot") + uri 
+    if isdir(absolute_path):
+        return handle_directory(absolute_path)
+    if isfile(absolute_path):
+        return handle_file(absolute_path)
+    raise NameError
```

Comment:
As I said in class, I like that you've pushed the handling of various types of endpoint into separate functions, makes them nicely testable.  Good work!